### PR TITLE
refactor: Standalone API

### PR DESCRIPTION
**Pull request by twisttaan**
### Todo:

- [ ] ~~Guilds~~
	- [ ] ~~Config~~
- [x] Render
	- [x] Stats
- [x] Stats
- [x] Tags
- [x] Users
	- [x] @me
- [x] Votes

### CODE REVIEW

**Comment by twisttaan** [comment]
Config was taken out of this PR since after this is merged I will be rewriting the way modules store configuration.

**Review by GodderE2D — COMMENTED** [review]
Not thoroughly reviewed, I'm not paid enough

File: services/api/src/routes/votes.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        !request.headers.authorization ||
+        request.headers.authorization !== getSecret("TOPGG_VOTE_SECRET")
+      )
+        return void response.status(403).send();
+
+      await postVote({
+        userSnowflake: request.body.user,
+        test: request.body.type === "test",
+        serviceName: "Top.gg",
+        voteLink: "https://top.gg/bot/807543126424158238/vote",
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply services/api/src/routes/votes.ts:25]: Evie shouldn't be self hosted for production use.

File: services/api/src/routes/votes.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        !request.headers.authorization ||
+        request.headers.authorization !== getSecret("DBL_VOTE_SECRET")
+      )
+        return void response.status(403).send();
+
+      await postVote({
+        userSnowflake: request.body.id,
+        test: false,
+        serviceName: "Discord Bot List",
+        voteLink: "https://discordbotlist.com/bots/jamble/upvote",
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply services/api/src/routes/votes.ts:47]: Evie shouldn't be self hosted for production use.

File: services/api/src/routes/stats.ts [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
   const { prisma } = container;
-  fastify.get("/", async (_, res) => {
-    const shardStats = await prisma.shardStats.findMany();
+  fastify.get<{
+    Querystring: {
+      avg?: string;
+    };
+  }>("/", async (req, res) => {
+    const stats =
+      req.query.avg === "true" || !req.query.avg
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply services/api/src/routes/stats.ts:13]: fixed in https://github.com/TeamEvie/Evie/pull/152/commits/f4874b111f5411a1d1a6094cce154430eb178fd3

File: services/api/src/routes/votes.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      type: "upvote" | "test";
+      isWeekend: boolean;
+      query: string;
+    };
+  }>(
+    "/topgg",
+
+    async (request, response) => {
+      if (
+        !request.headers.authorization ||
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply services/api/src/routes/votes.ts]: fixed in https://github.com/TeamEvie/Evie/pull/152/commits/f4874b111f5411a1d1a6094cce154430eb178fd3

File: services/api/src/routes/votes.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  fastify.post<{
+    Body: {
+      id: string;
+    };
+  }>(
+    "/dbl",
+
+    async (request, response) => {
+      if (
+        !request.headers.authorization ||
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply services/api/src/routes/votes.ts]: fixed in https://github.com/TeamEvie/Evie/pull/152/commits/f4874b111f5411a1d1a6094cce154430eb178fd3

File: services/api/src/routes/votes.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (GodderE2D) [reply services/api/src/routes/votes.ts:25]: At least tell people that maybe?

**Comment by twisttaan** [comment]
> Not thoroughly reviewed, I'm not paid enough

you don't get paid for open source 😔

**Comment by GodderE2D** [comment]
> > Not thoroughly reviewed, I'm not paid enough
> 
> you don't get paid for open source 😔

You didn't get my resignation letter the other day?

**Pull request merged**